jenkins/run-server-func-tests

```diff
@@ -75,11 +76,44 @@ function dump_journal {
     printf -- "\n--- journalctl dump ---\n\n"
 }
 
+function sc_disable {
+    # Run systemctl to disable the Pbench Server unit and clean up after any
+    # resulting failures attempting to do so.
+
+    # Send the command output to the bitbucket; capture stderr in a variable.
+    o=$(systemctl --user disable --now pbench-server 2>&1 >/dev/null)
+    r=${?}
+    if (( r != 0 )); then
+        # Ignore the expected error
+        if [[ ${o} != *"Unit file"*"does not exist"* ]]; then
+            echo "systemctl disable: ${o}"
+            return ${r}
+        fi
+    fi
+
+    # Even though we've disabled it above, the pbench-server service may still
+    # be present in systemctl's memory if it failed before or during being
+    # disabled.  A side-effect of the command below will purge this entry.
+    o=$(systemctl --user reset-failed pbench-server 2>&1 >/dev/null)
+    r=${?}
+    if (( r != 0 )); then
+        # Ignore the expected error
+        if [[ ${o} != *"Unit"*"not found"* && ${o} != *"Unit"*"not loaded"* ]]; then
+            echo "systemctl reset-failed: ${o}"
+            return ${r}
+        fi
+    fi
+
+    return ${r}
+}
+
```

Comment:
I wonder if the `undeploy` should be unbundled into a separate script as `deploy` was, so they're symmetric and can be used consistently...

Reply:
That's certainly worth pondering.  I'm not sure that there is much demand for an `undeploy` script outside the context of `run-server-func-tests`, but I suppose `Pipeline.gy` _might_ use it (although, I opted for a heavier hand, there).